### PR TITLE
Progress and Meter styling

### PR DESCRIPTION
**Pull request by rohmulan**
Some basic styling for progress bar. Still working on it. 

I would love some suggestions on making them visually look different a bit, I setup a [codepen](http://codepen.io/geekman-rohit/pen/qOGKMK)
Would like suggestions on a better file organization too.

### CODE REVIEW

**Comment by geekman-rohit** [comment]
Its based on #111 , will change and remove the commit if we decide not to use that.

File: demos/progress.html [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+<body>
+
+<h1>CSS Chassis</h1>
+<h2>Progress and Meter</h2>
+<hr>
+<h3>Meter Element</h3>
+<meter max="100" value="70" min="0" low="50" high="90"></meter>
+<meter max="100" value="40" min="20" low="50" high="90"></meter>
+<meter max="100" value="10" low="50" optimum="100" high="90"></meter>
+<meter class="ui_meter--striped " max="100" value="70" min="0" low="50" high="90"></meter>
```
Comment (cvrebert) [inline demos/progress.html]: Why do these 3 use underscores?

File: demos/progress.html [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+<meter class="ui_meter--striped " max="100" value="40" min="20" low="50" high="90"></meter>
+<meter class="ui_meter--striped " max="100" value="10" low="50" optimum="100" high="90"></meter>
+
+
+<h3>Meter Fallback</h3>
+<h5>Use ui-meter--modifier class for meter</h5>
+
+<div class="ui-meter--default">
```
Comment (cvrebert) [inline demos/progress.html]: The `<div>` versions should have ARIA attributes for accessibility; compare with http://getbootstrap.com/components/#progress-basic

**Comment by cvrebert** [comment]
I recommend against the approach of putting the percentage text within the progress bar itself.
(1) It results in inconsistency between `<div>`-based bars and `<progress>`-based bars (one has text, one doesn't)
(2) It breaks at low percentages; example: http://codepen.io/anon/pen/aOMKvZ . There's insufficient room for the text. And even if you tweak the position of the text, it might overlap both the filled and the unfilled portion of the progress bar; it is difficult to ensure that the text color will offer sufficient contrast against the background color in both situations.

As an alternative, I would suggest putting the text directly below the progress bar, and centered horizontally.

File: scss/molecules/progress/progress/_mixins.scss [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+@mixin ui-progress-fallback-stripes( $fgcolor, $bgcolor ) {
+	@include ui-progress-fallback( $fgcolor, $bgcolor );
+	span {
+		@include ui-stripes();
+	}
+	&--animated {
+		@include ui-progress-fallback( $fgcolor, $bgcolor );
+		span {
+			@include ui-stripes();
+			animation: ui-stripes-animate $ui-progress-animation-time linear infinite;
```
Comment (cvrebert) [inline scss/molecules/progress/progress/_mixins.scss]: Beware, this leaks memory in Chrome: http://crbug.com/429375

**Comment by kborchers** [comment]
Sorry for jumping in here but it seems like anything around what text and how it's displayed in a progress bar should be left up to a UI lib/framework built on top of Chassis. It feels too overbearing IMO to define that much detail but again, this is just my 2 cents so take it as such.

**Comment by arschmitz** [comment]
@kborchers I agree for the most part though i think we should offer some default styling.

@cvrebert I agree inside is troublesome i think we should opt for very basic styling either above or below what ever we use as a default style for any other label probably ( progress and meter elements are label-able elements ). For the most part unless its a percentage read out the text should probably be a label anyway.

Since we are using BEM that means inheritance is not an issue. That means its fairly trivial to provide default styles that don't matter if they are inside the toolbar or not. Then you can just put a span with the proper class where ever you like to get it properly styled ( and make sure your including proper aria attributes to get it properly associated )

**Comment by arschmitz** [comment]
sorry slipped wrong button DOH!

File: demos/progress.html (hunk 1 above)
Comment (geekman-rohit) [inline demos/progress.html]: sorry they shouldn't have underscores, I kinda forgot to change!

**Comment by cvrebert** [comment]
The pen's `<progress>` doesn't seem to work in Edge:
edge

**Comment by cvrebert** [comment]
[`:indeterminate`](https://developer.mozilla.org/en-US/docs/Web/CSS/:indeterminate) progress bars also need to be considered.

**Comment by geekman-rohit** [comment]
@cvrebert , I need to check how progress works on edge and make changes accordingly, thanks for pointing out!

**Comment by rohmulan** [comment]
Closing for now, until things move on with foms/redoing typography. will open/redo again once other things are done.

**Pull request closed without merging**